### PR TITLE
fix(api-server):  fix mmds PUT response status code

### LINKED ISSUE #528: MMDS - Return 204 instead of 201 on PUT requests
We don't send a body along with the status code in case of success, so we should use the status code 204 (NoContent). Currently the PUT /mmds request returns a  response with a null body and the status code 201.

### PR DESCRIPTION
**Pull request by fristonio**
* Fix response code of mmds PUT request to 204(No Content)
* Fixes #528

### CODE REVIEW

**Comment by andreeaflorescu** [comment]
Can you also update the Changelog please? I forgot to add this to the wiki. Because this is a customer facing issue, we should add a `Changed` section in which we say that PUT requests on /mmds always return 204 on success. Unfortunately I cannot give you an example as I am typing from my phone, but if you use git log on CHANGELOG.md you should see an example that has the Unreleased section in it.

**Comment by fristonio** [comment]
> Since we are not using the method is_initialized, we can delete it as well. It is defined in the mmds crate. Also, you should remove is_initialized field from the Mmds data structure.

`is_initialized` method of `Mmds` structure is also used in `patchMmds`(api-server) so I don't think we can remove that. Here https://github.com/firecracker-microvm/firecracker/blob/master/api_server/src/http_service.rs#L485

**Comment by fristonio** [comment]
@alexandruag I have squashed the commits and did a force push.
I will take care of this next time.

**Comment by fristonio** [comment]
I am not sure why the checks are failing.
@andreeaflorescu can you take a look at this once.

**Comment by raduweiss** [comment]
Looking at the error it doesn't seem connected to your changes (see below). Still, just to make sure, did the tests pass on your side for you before submitting the PR?

Here's what we see in our CI:

```
failures:

---- tests::test_machine_configuration stdout ----
thread 'tests::test_machine_configuration' panicked at 'Cannot add write metrics TimerFd to epoll.: EpollFd(Os { code: 9, kind: Other, message: "Bad file descriptor" })', libcore/result.rs:1009:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
1: std::sys_common::backtrace::print
at libstd/sys_common/backtrace.rs:71
at libstd/sys_common/backtrace.rs:59
2: std::panicking::default_hook::{{closure}}
at libstd/panicking.rs:211
3: std::panicking::default_hook
at libstd/panicking.rs:221
4: std::panicking::rust_panic_with_hook
at libstd/panicking.rs:477
5: std::panicking::continue_panic_fmt
at libstd/panicking.rs:391
6: rust_begin_unwind
at libstd/panicking.rs:326
7: core::panicking::panic_fmt
at libcore/panicking.rs:77
8: core::result::unwrap_failed
at libcore/macros.rs:26
9: <core::result::Result<T, E>>::expect
at libcore/result.rs:835
10: vmm::Vmm::new
at vmm/src/lib.rs:562
11: vmm::tests::create_vmm_object
at vmm/src/lib.rs:1850
12: vmm::tests::test_machine_configuration
at vmm/src/lib.rs:2022
13: vmm::tests::test_machine_configuration::{{closure}}
at vmm/src/lib.rs:2021
14: core::ops::function::FnOnce::call_once
at libcore/ops/function.rs:238
15: <F as alloc::boxed::FnBox<A>>::call_box
at libtest/lib.rs:1461
at libcore/ops/function.rs:238
at liballoc/boxed.rs:646
16: __rust_maybe_catch_panic
at libpanic_unwind/lib.rs:103
```

**Comment by andreeaflorescu** [comment]
Yap, my bad. In that case, can you update the docstring of the is_initialized method? It is not accurate anymore.

**Comment by raduweiss** [comment]
@fristonio looks like the issue causing that failure is unrelated to your PR. I've cut #706 for it.

**Comment by fristonio** [comment]
This time the tests were successful. @andreeaflorescu does this look good to you?

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@
 - Moved the `instance_info_fails` metric under `get_api_requests`
 - Improved [readme](README.md) and added links to more detailed information,
   now featured in subject-specific docs.
+- PUT requests on /mmds always return 204 on success.
```
Comment (dianpopa) [inline CHANGELOG.md]: The changes made after the last release should be added to a new section called 'Unreleased' (checkout [How to Keep a CHANGELOG](https://keepachangelog.com/en/0.3.0/) ):
So it should look something like this:
```markdown
# Changelog

## Unreleased

### Changed
- PUT requests on /mmds always return 204 on success.

## [0.11.0]

### Added
...
```

Comment (dianpopa) [inline CHANGELOG.md]: The changes made after the last release should be added to a new section called 'Unreleased' (checkout [How to Keep a CHANGELOG](https://keepachangelog.com/en/0.3.0/) ):
So it should look something like this:
```markdown
# Changelog

## Unreleased

### Changed
- PUT requests on /mmds always return 204 on success.

## [0.11.0]

### Added
...
```

Comment (fristonio) [reply CHANGELOG.md]: I have updated the pull request, will take care of this the next time.

**Pull request merged**